controllers/chat/sendAnonymousMessage.js

```diff
@@ -65,6 +67,23 @@ const sendAnonymousMessage = async (req, res) => {
     if (createdChannel?.channel?.is_channel_blocked)
       return res.status(403).json(responseError('Channel is blocked'));
 
+    // Prevent if user try to chat with blocked user
+    let channelType = createdChannel?.channel?.channel_type;
+    if (channelType === 4 || channelType === 0) {
+      let signUser = await UsersFunction.findSignedUserId(User, req.userId);
+      const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, signUser);
+      let better_channel_member = createdChannel?.channel?.better_channel_member;
+      if (blockedIds.length > 0) {
+        for (const member of better_channel_member) {
+          if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
+            return res
+              .status(403)
+              .json(responseError('This user does not want to receive messages'));
+          }
+        }
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user is blocked seems to be correct. However, it could be optimized by reducing the complexity of the code. Currently, you are using a loop to check if a member is in the blocked list which results in O(n) time complexity. This can be improved by converting the `blockedIds` array into a Set and then checking if a member exists in that set. This would reduce the time complexity to O(1).

Here's how you can do it:

```diff
-      if (blockedIds.length > 0) {
-        for (const member of better_channel_member) {
-          if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
-            return res
-              .status(403)
-              .json(responseError('This user does not want to receive messages'));
-          }
-        }
-      }
+      const blockedIdsSet = new Set(blockedIds);
+      const isBlocked = better_channel_member.some(member => member.user_id !== req.userId && blockedIdsSet.has(member.user_id));
+      if (isBlocked) {
+        return res
+          .status(403)
+          .json(responseError('This user does not want to receive messages'));
+      }
```
This change will improve the performance of your code when dealing with large lists of blocked users.


<!-- This is an auto-generated comment by OSS CodeRabbit -->